src/diff/children.js

```diff
@@ -123,6 +123,11 @@ export function diffChildren(
 						firstChildDom = newDom;
 					}
 
+					if (excessDomChildren != null) {
+						const index = excessDomChildren.indexOf(newDom);
+						if (index !== -1) excessDomChildren[index] = null;
```

Comment:
Do you think we would run into issue -- deopting and such, if we always set the index, whatever the value? 🤔 
```suggestion
						excessDomChildren[excessDomChildren.indexOf(newDom)] = null;
```